src/test/java/com/diffblue/deeptestutils/mock/MethodMatcherTest.java

```diff
@@ -0,0 +1,277 @@
+package com.diffblue.deeptestutils.mock;
+
+import org.junit.Assert;
+import org.junit.rules.ExpectedException;
+import org.junit.Test;
+
+import java.lang.reflect.Array;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+import java.lang.reflect.InvocationTargetException;
+
+class Y extends com.diffblue.deeptestutils.mock.subpackage.X<Integer> {
+  protected Integer foo() {
+    return 42;
+  }
+}
+
+interface I {
+  public Number foo();
+}
+
+class Z extends Y implements I {
+  public Integer foo() {
+    return 47;
+  }
+}
+
+class Parent {
+  public String toString() {
+    return val;
+  }
+  public String val;
+  public Parent(String val) {
+    this.val = val;
+  }
+
+  static Parent simple(Parent a) {
+    return new Parent("0");
+  }
+
+  static Parent returnSubtype(Parent a) {
+    return new Parent("2");
+  }
+
+  static Parent argSubtype(Parent a) {
+    return new Parent("4");
+  }
+
+  static Parent argSupertypeReturnSubtype(Child a) {
+    return new Parent("6");
+  }
+
+  static Parent mix1(Child a) {
+    return new Parent("8");
+  }
+}
+
+class Child extends Parent{
+  public Child(String val) {
+     super(val);
+  }
+
+  static Parent simple(Parent a) {
+   return new Parent("1");
+  }
+
+  static Child returnSubtype(Parent a) {
+    return new Child("3");
+  }
+
+  static Parent argSubtype(Child a) {
+    return new Parent("5");
+  }
+
+  static Child argSupertypeReturnSubtype(Parent a) {
+    return new Child("7");
+  }
+
+  static Parent mix1(Child a) {
+    return new Child("9x");
+  }
+
+  static Child mix1(Parent a) {
+    return new Child("9");
+  }
+}
+
+public class MethodMatcherTest {
+
+  @org.junit.Rule
+  public ExpectedException thrown = ExpectedException.none();
+
+  @org.junit.Test
+  public void simpleTest() throws IllegalAccessException, InvocationTargetException {
+    Method m1 = DTUMemberMatcher.method(Child.class, "simple", Parent.class);
+    Method m2 = DTUMemberMatcher.methodUsingWholeSignature(Child.class,"simple", Parent.class);
+
+    Child child = new Child("child");
+    Parent parent = new Parent("parent");
+
+    String s1 = m1.invoke(child, parent).toString();
+    String s2 = m2.invoke(child, parent).toString();
+
+    // verify that we find the same method using both DTUMemberMatcher methods
+    Assert.assertEquals(s1, s2);
```

Comment:
`method` invokes `methodUsingWholeSignature` directly (with the same arguments), so I don't see the point of this check. :hamburger: 
From my point of view this is a weak and unnecessary check so we could remove it (thus making these tests more robust to implementation changes) and that would allow to remove the `m2` definitions (in all the tests) and make `methodUsingWholeSignature` private as it should be.